onnxruntime/python/onnxruntime_pybind_state_common.h

```diff
@@ -5,19 +5,82 @@
 #include "core/common/logging/sinks/cerr_sink.h"
 #include "core/framework/allocator.h"
 #include "core/framework/session_options.h"
-
 #include "core/session/environment.h"
+#include "core/session/inference_session.h"
 
 namespace onnxruntime {
-class InferenceSession;
-
 namespace python {
 
 using namespace onnxruntime;
 using namespace onnxruntime::logging;
 
-inline const SessionOptions& GetDefaultCPUSessionOptions() {
-  static SessionOptions so;
+struct CustomOpLibrary {
+  CustomOpLibrary(const char* library_path, OrtSessionOptions& ort_so);
+
+  ~CustomOpLibrary();
+
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(CustomOpLibrary);
+
+ private:
+  void UnloadLibrary();
+
+  std::string library_path_;
+  void* library_handle_ = nullptr;
+};
+
+// Thin wrapper over internal C++ SessionOptions to accommodate custom op library management for the Python user
+struct PySessionOptions : public SessionOptions {
+  // `PySessionOptions` has a vector of shared_ptrs to CustomOpLibrary, because so that it can be re-used for all
+  // `PyInferenceSession`s using the same `PySessionOptions` and that each `PyInferenceSession` need not construct
+  // duplicate CustomOpLibrary instances.
+  std::vector<std::shared_ptr<CustomOpLibrary>> custom_op_libraries_;
+
+  // Hold raw `OrtCustomOpDomain` pointers - it is upto the shared library to release the OrtCustomOpDomains
+  // that was created when the library is unloaded
+  std::vector<OrtCustomOpDomain*> custom_op_domains_;
+};
+
+// Thin wrapper over internal C++ InferenceSession to accommodate custom op library management for the Python user
+struct PyInferenceSession {
+  // Default ctor is present only to be invoked by the PyTrainingSession class
+  explicit PyInferenceSession() {}
+
+  explicit PyInferenceSession(Environment& env, const PySessionOptions& so, const std::string& arg, bool is_arg_file_name) {
```

Comment:
explicit constructors make sense only for single parameter constructors. no need for explicit here.

Reply:
Fixed